src/main/java/com/conveyal/datatools/manager/models/Deployment.java

```diff
@@ -417,6 +420,31 @@ public void dump (File output, boolean includeManifest, boolean includeOsm, bool
         out.close();
     }
 
+    /**
+     * Determine the entry name for a GTFS file within the deployment bundle.
+     * This prioritizes the FeedSource filename if available and valid, otherwise falls back
+     * to the original GTFS filename derived from the FeedVersion.
+     *
+     * @param feedVersion The FeedVersion being processed.
+     * @param gtfsFile    The GTFS file associated with the FeedVersion.
+     * @return The calculated entry name for the zip file.
+     */
+    public String getFeedSourceBundleFilename(FeedVersion feedVersion, File gtfsFile) {
+        String entryName;
+        FeedSource fs = feedVersion.parentFeedSource();
+
+        if (fs != null && !Strings.isBlank(fs.filename)) {
+            // Use FeedSource filename if available, ensuring it ends with .zip
+            entryName = fs.filename.endsWith(".zip") ? fs.filename : fs.filename + ".zip";
+            LOG.info("Using FeedSource filename for zip entry: {}", entryName);
+        } else {
+            // Fallback to the original GTFS filename derived from FeedVersion
+            entryName = gtfsFile.getName();
+            LOG.info("Using FeedVersion filename for zip entry: {}", entryName);
```

Comment:
Nit: you could log first then `return` from inside the if/else block, and remove lines 434 and 445.